tests/org.pitest.pitclipse.ui.tests/src/org/pitest/pitclipse/ui/tests/PitclipsePitSummaryViewTest.java

```diff
@@ -0,0 +1,84 @@
+
+import org.eclipse.core.runtime.CoreException;
+import org.eclipse.swtbot.swt.finder.junit.SWTBotJunit4ClassRunner;
+import org.eclipse.swtbot.swt.finder.utils.SWTBotPreferences;
+import org.junit.After;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.pitest.pitclipse.ui.behaviours.pageobjects.PitSummaryView;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.endsWith;
+import static org.hamcrest.Matchers.not;
+import static org.pitest.pitclipse.ui.behaviours.pageobjects.PageObjects.PAGES;
+
+import java.io.File;
+
+/**
+ * @author Jonas Kutscha
+ * 
+ */
+@RunWith(SWTBotJunit4ClassRunner.class)
+public class PitclipsePitSummaryViewTest extends AbstractPitclipseSWTBotTest {
+    private static final String TEST_PROJECT = "org.pitest.pitclipse.testprojects.twoclasses";
+    private static final String FOO_BAR_PACKAGE = "foo.bar";
+    private static final String FOO_CLASS = "Foo";
+    private static final String TEST_PROJECT_WITH_DIFFERENT_SRC = "org.pitest.pitclipse.testprojects.differentSrc";
+    private static final String FOO_PACKAGE = "foo";
+    private static PitSummaryView summaryView;
+
+    @BeforeClass
+    public static void setupJavaProject() throws CoreException {
+        summaryView = PAGES.getPitSummaryView();
+        importTestProject(TEST_PROJECT);
+        importTestProject(TEST_PROJECT_WITH_DIFFERENT_SRC);
+    }
+
+    @After
+    public void resetView() {
+        summaryView.resetView();
+        bot.closeAllEditors();
+    }
+
+    @Test
+    public void openPitSummaryViewAtCorrectUrl() throws CoreException {
+        runPackageTest(FOO_BAR_PACKAGE, TEST_PROJECT);
+        coverageReportGenerated(2, 80, 0);
+        assertThat(summaryView.getCurrentBrowserUrl(), endsWith("index.html"));
+        assertThat(summaryView.getCurrentBrowserUrl(),
+                not(endsWith(File.separatorChar + "foo" + File.separatorChar + "index.html")));
+        assertThat(summaryView.getCurrentBrowserUrl(), not(
+                endsWith(File.separatorChar + "foo" + File.separatorChar + "bar" + File.separatorChar + "index.html")));
+
+        openEditor(FOO_CLASS, FOO_PACKAGE, TEST_PROJECT_WITH_DIFFERENT_SRC);
+        runPackageTest(FOO_PACKAGE, TEST_PROJECT_WITH_DIFFERENT_SRC);
+        coverageReportGenerated(1, 80, 0);
+        assertThat(summaryView.getCurrentBrowserUrl(), endsWith("foo" + File.separatorChar + "Foo.java.html"));
+    }
+
+    @Test
+    public void navigateWithButtons() throws CoreException {
+        final long timeout = SWTBotPreferences.TIMEOUT;
+        openEditor(FOO_CLASS, FOO_PACKAGE, TEST_PROJECT_WITH_DIFFERENT_SRC);
+        runPackageTest(FOO_PACKAGE, TEST_PROJECT_WITH_DIFFERENT_SRC);
+        coverageReportGenerated(1, 80, 0);
+        // set timeout to 5 ms, because the browser should not load pages and run into
+        // timeout
+        SWTBotPreferences.TIMEOUT = 5;
```

Comment:
If you change such a value, you must make sure you always reset it, even if the test fails (or this will break other subsequent tests): either you use a try...finally block (and you reset it in the finally block) or you use `@Before` and `@After`